docs/pages/access-controls/guides/device-trust.mdx

```diff
@@ -0,0 +1,326 @@
+user/device combination.
+
+<Admonition type="tip" title="Enrollment">
+  Device enrollment is key in correctly identifying a device throughout its
+  lifetime. The more trusted the enrollment operator, the better are the ongoing
+  guarantees that the device itself is trustworthy.
+</Admonition>
+
+The tie between the device inventory and enrollment is a device enrollment
+token. A device enrollment token needs to be created by a device admin and sent
+to the person performing the enrollment ceremony. The token ties the actual
+device being enrolled to the expected device in the inventory, adding an extra
+layer of protection.
+
+## Prerequisites
+
+(!docs/pages/includes/commercial-prereqs-tabs.mdx!)
+
+- A signed and notarized `tsh` binary for enrollment.
+  [Download the macOS tsh installer](../../installation.mdx#macos).
+- A macOS device to register and enroll.
+
+## Step 1/2. Register a trusted device
+
+The `tctl` tool is used to manage the device inventory. A device admin is
+responsible for managing devices, adding new devices to the inventory and
+removing devices that are no longer in use.
+
+A role that grants permissions for the `device` resource is necessary to manage
+the inventory. Save the yaml below as `device-admin.yaml` and create it in your
+cluster:
+
+```yaml
+version: v6
+kind: role
+metadata:
+  name: device-admin
+spec:
+  allow:
+    rules:
+    - resources: ["device"]
+      verbs:
+      - create
+      - read
+      - list
+      - update
+      - delete
+      - create_enroll_token
+      - enroll
+```
+
+```code
+$ tctl create -f device-admin.yaml
+# role 'device-admin' has been created
+```
+
+<Admonition type="tip" title="Editor Role">
+  For clusters created using Teleport v12 or newer, the builtin role `editor`
+  has the necessary permissions to manage devices.
+  For production setups a separate role is still recommended.
+</Admonition>
+
+Note that, in addition to the usual CRUD verbs (create, read, list, update and
+delete), we have also included `create_enroll_token` and `enroll`. The
+`create_enroll_token` verb is necessary to execute the `tctl devices enroll`
+command; `enroll` is necessary to execute `tsh device enroll`. Both are
+demonstrated in the following section.
+
+Make sure your user has the `device-admin` role:
+
+```yaml
+kind: user
+metadata:
+  name: codingllama
+  # unrelated fields omitted.
+spec:
+  roles:
+  - access
+  - device-admin # add this line
+version: v2
+```
+
+```code
+$ tctl edit users/codingllama
+# user "codingllama" has been updated
+$ tsh logout; tsh login --proxy=example.com --user=codingllama ## fetch new certificates
+```
+
+Having the necessary permissions, let's add a device to the inventory. Replace
+the `$serial` below with your macOS device serial number (visible under Apple ->
+"About This Mac" -> "Serial number").
+
+```code
+$ serial="$(ioreg -c IOPlatformExpertDevice -d 2 | grep -i IOPlatformSerialNumber | awk -F'"' '{print $4}')"
```

Comment:
Above we tell the reader to replace `$serial` (hence my earlier comment) but here we're explicitly defining it.

Should we make it an uppercase var to follow convention?

Reply:
Yeah, I didn't like this one too much, but it's important to explain what the asset tag is and the command is kinda hard to parse.

I've added a nod to the script in the parenthesis above.

> Should we make it an uppercase var to follow convention?

Done. Do we have our conventions documented somewhere? Browsing other docs for examples is often proving unreliable to me.

Reply:
The convention I'm referring to isn't our own.

[Reference 1](https://www.tutorialspoint.com/unix/unix-using-variables.htm#:~:text=The%20name%20of%20a%20variable,have%20their%20names%20in%20UPPERCASE.)

[Reference 2](https://stackoverflow.com/questions/673055/correct-bash-and-shell-script-variable-capitalization)